app/diagnostics.go

```diff
@@ -1110,3 +1061,128 @@ func (s *Server) trackChannelModeration() {
+
+	// knownPlugins maps plugin IDs to telemetry suffixes
+	var knownPlugins = map[string]string{
+		"antivirus":                                   "antivirus",
+		"mattermost-autolink":                         "autolink",
+		"com.mattermost.aws-sns":                      "aws_sns",
+		"com.mattermost.confluence":                   "confluence",
+		"com.mattermost.custom-attributes":            "custom_user_attributes",
+		"github":                                      "github",
+		"com.github.manland.mattermost-plugin-gitlab": "gitlab",
+		"jenkins":                                     "jenkins",
+		"jira":                                        "jira",
+		"jitsi":                                       "jitsi",
+		"com.mattermost.mscalendar":                   "mscalendar",
+		"com.mattermost.nps":                          "nps",
+		"skype4business":                              "skype4business",
+		"com.mattermost.plugin-todo":                  "todo",
+		"com.mattermost.webex":                        "webex",
+		"com.mattermost.welcomebot":                   "welcome_bot",
+		"zoom":                                        "zoom",
+	}
+
+	marketplacePlugins, err := s.getAllMarketplaceplugins(cfg)
+	if err != nil {
+		log.Println("keks 1")
+		mlog.Info("Failed to fetch marketplace plugins for telemetry. Using predefined list.", mlog.Err(err))
+
+		for k, v := range knownPlugins {
+			pluginConfigData["enable_"+v] = pluginActivated(cfg.PluginSettings.PluginStates, k)
+		}
+	} else {
+		log.Println("keks 2")
+		for _, p := range marketplacePlugins {
+			id := p.Manifest.Id
+			var key string
+
+			// Old old naming schema for to keep consistency with old telemetry data
+			if v, ok := knownPlugins[id]; ok {
+				key = "enable_" + v
+			} else {
+				key = "enable_" + id
+			}
+
+			pluginConfigData[key] = pluginActivated(cfg.PluginSettings.PluginStates, id)
+		}
+	}
+
+	pluginsEnvironment := s.GetPluginsEnvironment()
+	if pluginsEnvironment != nil {
+		if plugins, appErr := pluginsEnvironment.Available(); appErr != nil {
+			mlog.Error("Unable to add plugin versions to diagnostics", mlog.Err(appErr))
+		} else {
+
+			// If marketplace request failed, use predefined list
+			if marketplacePlugins == nil {
+				for k, v := range knownPlugins {
+					pluginConfigData["version_"+v] = pluginVersion(plugins, k)
+				}
+			} else {
+				for _, p := range marketplacePlugins {
+					id := p.Manifest.Id
+					var key string
+
+					// Old old naming schema for to keep consistency with old telemetry data
+					if v, ok := knownPlugins[id]; ok {
+						key = "version_" + v
+					} else {
+						key = "version_" + id
+					}
+
+					pluginConfigData[key] = pluginVersion(plugins, id)
+				}
+			}
+		}
+	}
+
+	s.SendDiagnostic(TRACK_CONFIG_PLUGIN, pluginConfigData)
+}
+
+func (s *Server) getAllMarketplaceplugins(cfg *model.Config) ([]*model.BaseMarketplacePlugin, error) {
+	marketplaceClient, err := marketplace.NewClient(
+		*cfg.PluginSettings.MarketplaceUrl,
```

Comment:
I was expecting we would unconditionally query from /our/ marketplace here, not the configured one. Otherwise we're using the customer's list of private plugins as an authority for the whitelist, which I'm assuming is not desired.

Reply:
Yes, I think that's correct (that we do NOT want to implicitly track the customer's list of private plugins).  @hanzei Perhaps I misunderstood something.

I think we should track the number of plugins installed/enabled on a server as a count.  But only have data about version, etc. From plugins that are within the MM marketplace.

Reply:
Got it. Thanks for the clarification. Changed the code to always fetch the list from https://api.integrations.mattermost.com.

> I think we should track the number of plugins installed/enabled on a server as a count. 

We already track that data in `enabled_plugins`.